src/Extend/Frontend.php

```diff
@@ -236,6 +272,24 @@ function (ActualFrontend $frontend, Container $container) {
         );
     }
 
+    private function registerPreloads(Container $container): void
+    {
+        if (empty($this->preloads)) {
+            return;
+        }
+
+        $container->resolving(
+            "flarum.frontend.$this->frontend",
+            function (ActualFrontend $frontend, Container $container) {
+                $frontend->content(function (Document $document) use ($container) {
+                    foreach ($this->preloads as $preload) {
+                        $document->preloads[] = is_callable($preload) ? ContainerUtil::wrapCallback($preload, $container)($document) : $preload;
+                    }
```

Comment:
This logic seems flawed. Shouldn't we call the callback here instead of just wrapping (since we're already in `resolving`), and flatten the preloads as we load them into the document?

Reply:
we are calling the callback here